plugins/Goals/Visualizations/GoalsPages.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\Goals\Visualizations;
+
+use Piwik\API\Request;
+use Piwik\Common;
+use Piwik\DataTable\Filter\AddColumnsProcessedMetricsGoal;
+use Piwik\Piwik;
+use Piwik\Plugins\CoreVisualizations\Visualizations\HtmlTable;
+use Piwik\Site;
+
+require_once PIWIK_INCLUDE_PATH . '/core/Twig.php';
+
+/**
+ * DataTable Visualization that derives from HtmlTable and shows goals metrics for page conversions
+ */
+class GoalsPages extends Goals
```

Comment:
I was wondering if those new Visualizations are needed. Overwriting some config variables should also be possible in the report classes. Did you maybe try that already and had any problems with it?
Generally there wouldn't be a problem with having additional visualizations, but there are a couple of checks for the `tableGoal` type in JS, that won't work for this new types automatically. e.g.
https://github.com/matomo-org/matomo/blob/b7bfb36a3d37e4d056ab3f0355672b85f318f3d9/plugins/CoreHome/javascripts/dataTable.js#L1975
https://github.com/matomo-org/matomo/blob/6b6f572693e1e4c0d0b371133bd2082e0c09525a/plugins/CoreHome/javascripts/dataTable_rowactions.js#L329
and maybe others

Reply:
The goals visualization adds the wrong goal metric columns to the report, I did initially try to set the correct metric columns via the report but the visualization then adds standard columns too. I've updated any javascript I could find that references the `tableGoals` visualization to also work with `tableGoalsEntryPages` and `tableGoalsPages`